DEV.md

```diff
@@ -0,0 +1,191 @@
+│   └── xpi.sh            # build the XPI
+├── dist                  # built xpis (addons)
+│   ├── .gitignore
+│   ├── @template-button-study.shield.mozilla.com-1.2.0.xpi
+│   └── linked-addon.xpi -> @template-button-study.shield.mozilla.com-1.2.0.xpi
+├── package-lock.json
+├── package.json
+├── run-firefox.js        # used by `npm run firefox`
+├── survival.md
+├── templates             # mustache templates, filled from `package.json`
+│   ├── chrome.manifest.mustache
+│   └── install.rdf.mustache
+└── test                  # Automated tests `npm test` and circle
+│   ├── Dockerfile
+│   ├── docker_setup.sh
+│   ├── functional_tests.js
+│   ├── test_harness.js
+│   ├── test_printer.py
+│   └── utils.js
+└── tutorial.md
+
+>> tree -a -I 'node_modules|.git|.DS_Store'
+```
+
+This structure is set forth in [shield-studies-addon-template](https://github.com/mozilla/shield-studies-addon-template), with study-specific changes found mostly in `addon/lib`, `addon/webextension` and `addon/Config.jsm`.
+
+## General Shield Study Engineering
+
+Shield study add-ons are legacy (`addon/bootstrap.js`) add-ons with an optional embedded web extension (`addon/webextension/background.js`).
+
+The web extension needs to be packaged together with a legacy add-on in order to be able to access Telemetry data, user preferences etc that are required for collecting relevant data for [Shield Studies](https://wiki.mozilla.org/Firefox/Shield/Shield_Studies).
+
+It is recommended to build necessary logic and user interface using in the context of the webextension and communicate with the legacy add-on code through messaging whenever privileged access is required.
+
+For more information, see [./about.md]
+
+### Description of what goes on when this add-on is started
+
+During `bootstrap.js:startup(data, reason)`:
+
+    a. `shieldUtils` imports and sets configuration from `Config.jsm`
+    b. `bootstrap.js:chooseVariation` explicitly and deterministically chooses a variation from `studyConfig.weightedVariations`
+    c.  the WebExtension starts up
+    d.  `boostrap.js` listens for requests from the `webExtension` that are study related:  `["info", "telemetry", "endStudy"]`
+    e.  `webExtension` (`background.js`) asks for `info` from `studyUtils` using `askShield` function.
+    f.  Feature starts using the `variation` from that info.
+    g.  Feature instruments user button to send `telemetry` and to `endStudy` if the button is clicked enough.
+
+Tip: For more insight on what is study-specific, compare the source code of previously deployed shield studies with this template (and each other) to get an idea of what is actually relevant to change between studies vs what is mostly untouched boilerplate.
+
+### Legacy repositories
+
+Repositories that should no longer be used as templates for new studies:
+
+[https://github.com/gregglind/template-shield-study]() - The incubation repo for the updated structure and contents of this repo, implemented in late 2017.
```

Comment:
This looks "off". Not sure if we should duplicate the URL in the parens, or just use the `<https://github.com/gregglind/template-shield-study>` syntax since the URL and label are the same.

Same w/ the hello world example link below.